### DISCUSSION TITLE
Can't read rom header on SNES or GB/GBC

### DISCUSSION THREAD

**Discussion by PranceRosner** [sanni/cartreader#687] [Need help]
I just competed a build on the rev 5 v3 hardware, flashed with v12 beta.  I have been getting rom header errors for SNES and GB/GBC carts only. I've tried multiple carts, cleaning contacts, etc. without success.  NES, Famicom, GBA, N64, Genesis, and SMS carts all work.  Is there anything I should check here?

**Comment by sanni** [comment]
That GBA works and GB doesn't is a little weird since they use the same pins and slot. Can you check if the cartridges are originals or China fakes?

What you can do is ignore the error, read the Gameboy ROM anyway, copy it to your PC and then open the faulty ROM with a hex editor like [Nexus HxD](https://mh-nexus.de/en/hxd).
And then post a screenshot of HxD here. Usually by comparing the faulty dump to a known good dump you can narrow the issue down.

**Reply by PranceRosner** [reply]
Thanks for taking a look.  These are definitely authentic, I'm the original owner since the 90s!  Here is a dump of Super Mario Land, it gave me a header checksum error. This is exactly what I've seen on any other GB cart.

**Reply by sanni** [reply]
Looks like some of the data pins don't have a good connection or are shorted to other pins.
Re-solder everything marked in these images:

![efrwrsffffwfw](https://user-images.githubusercontent.com/19169494/213728849-e6f96d65-39fa-4d57-bc75-a3d2e35cc0d8.jpg)

![fwaffqfaffff](https://user-images.githubusercontent.com/19169494/213729595-9bee0bf1-4148-4ad0-a483-7af22b34f212.jpg)

These you have to re-solder both on the main PCB and the Mega 2560 Pro
![rgsgsesgsgsggs](https://user-images.githubusercontent.com/19169494/213728853-84c3e133-d0a4-4bde-aa4f-0d90c0f95f39.jpg)

Then try dumping GB again without any other slot PCB attached to the main PCB.

**Reply by PranceRosner** [reply]
So after some careful scrutiny, I was checking the Arduino I purchased and found several bridged pins. I think this is the culprit... I will clean up the bridges and report back soon.

![image](https://user-images.githubusercontent.com/123119646/213742590-45dd9600-2be4-4d00-9b6a-af1c64f38dd7.png)

**Comment by sanni** [comment]
On the Atmega the GB D0 to D7 data pins would be those pins:
![vdsfvssvdvsvs](https://user-images.githubusercontent.com/19169494/213743172-593d976f-4ca7-46d8-a39b-f236244041d5.jpg)

That does indeed correspond to the bridges you found.

**Reply by PranceRosner** [reply]
Alright, it's reading headers on games correctly now. I was able to dump several successfully on both GB and SNES. However, I decided to throw an SA1 cart at it, Super Mario RPG.  It read the header correctly, but dumping it was not successful.  Here's the log and a hex editor screenshot. Thanks again for the support here!

OSCR HW5 V12.0

[+] Super Nintendo/SFC

[+] SNES/SFC cartridge


Searching database...
Checksum: 3BB4
Header CRC32: F2DBCF9A
Found

Title: SUPER MARIO RPG
Revision: 0
Type: HiROM SlowROM
ICs: ROM SA1 RAM BATT
ROM Size: 4 MB (64 banks)
Save Size: 32 KB
Checksum: 3BB4
Press Button...

[+] Read ROM


Saving to SNES/ROM/SUPER MARIO RPG/14/...
Dumping HiRom...
[*******************]
Checksum... B007 != 3BB4
Invalid Checksum
CRC32... FEEF7753 -> Not found

![image](https://user-images.githubusercontent.com/123119646/213756348-fed398f7-cc59-44b7-8402-8e6c34f4f9e8.png)

**Reply by PranceRosner** [reply]
Disregard, I poked around online and several folks said you need to _REALLY_ clean your carts well. It's dumping fine now, and anything I throw at it is working 100%.  Again, thank you for the support, it's a great project and I look forward to what the community can continue to add to it!
